### PR TITLE
[clang-tidy] modernize-use-nullptr

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: include/openthread/coap.h [hunk 1]
```diff
@@ -826,7 +826,7 @@ static inline otError otCoapSendRequest(otInstance *          aInstance,
                                         otCoapResponseHandler aHandler,
                                         void *                aContext)
 {
-    return otCoapSendRequestWithParameters(aInstance, aMessage, aMessageInfo, aHandler, aContext, NULL);
+    return otCoapSendRequestWithParameters(aInstance, aMessage, aMessageInfo, aHandler, aContext, nullptr);
```
Comment (wgtdkp) [inline include/openthread/coap.h:829]: `nullptr` is C++ only while the public headers could be included in C sources.

File: tests/unit/test_platform.cpp [hunk 2]
```diff
@@ -302,7 +302,7 @@ otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
     }
     else
     {
-        return (otRadioFrame *)0;
+        return (otRadioFrame *)nullptr;
```
Comment (simonlingoogle) [inline tests/unit/test_platform.cpp:305]:
```suggestion
        return nullptr;
```
nit: is the type conversion necessary?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with a nit

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

File: include/openthread/coap.h [hunk 3]
```diff
@@ -909,7 +910,7 @@ otError otCoapSendResponseWithParameters(otInstance *              aInstance,
  */
 static inline otError otCoapSendResponse(otInstance *aInstance, otMessage *aMessage, const otMessageInfo *aMessageInfo)
 {
-    return otCoapSendResponseWithParameters(aInstance, aMessage, aMessageInfo, NULL);
+    return otCoapSendResponseWithParameters(aInstance, aMessage, aMessageInfo, nullptr);
```
Comment (wgtdkp) [inline include/openthread/coap.h:913]: ditto (sorry I didn't mention this one).

**Review by Irving-cl — APPROVED** [review]
LGTM

**Review by abtink — APPROVED** [review]
LGTM 👍

**Pull request merged**